homeassistant/components/here_travel_time/sensor.py

```diff
@@ -309,10 +316,34 @@ def icon(self) -> str:
             )
             return self._get_location_from_attributes(zone_entity)
 
-        # If zone was not found in state then use the state as the location
-        if entity_id.startswith("sensor."):
+        # Resolve nested entity
+        if recursion_history is None:
+            recursion_history = []
+        recursion_history.append(entity_id)
+        if entity.state in recursion_history:
+            _LOGGER.error(
+                "Circular Reference detected. The state of %s has already been checked.",
+                entity.state,
+            )
+            return None
+        _LOGGER.debug("Getting nested entity for state: %s", entity.state)
+        nested_entity = self.hass.states.get(entity.state)
+        if nested_entity is not None:
+            _LOGGER.debug("Resolving nested entity_id: %s", entity.state)
+            return await self._get_location_from_entity(entity.state, recursion_history)
+
+        # Check if state is valid coordinate set
+        pattern = r"-?\d{1,2}\.\d+,-?\d{1,3}\.\d+"
+        if re.fullmatch(pattern, entity.state):
```

Comment:
I would prefer if we use our validators. They are more clear.

https://github.com/home-assistant/home-assistant/blob/a3c98440e0e208a162f90ad67a5f843e5011f153/homeassistant/helpers/config_validation.py#L70